File: output/gemma4-e2b/avatar/Java/Python/codeforces_596_B.py
import sys

def solve():
    # Read all input from stdin
    data = sys.stdin.read().split()
    
    if not data:
        return

    # The first number is N
    n = int(data[0])
    
    if n == 0:
        print(0)
        return

    # The remaining numbers are the array elements
    arr = [int(x) for x in data[1:]]
    
    # Calculate the result based on the Java logic: |a_0| + sum(|a_i - a_{i-1}|)
    
    # Initialize answer with |arr[0]|
    answer = abs(arr[0])
    
    # Add the sum of absolute differences between adjacent elements
    for i in range(1, n):
        answer += abs(arr[i] - arr[i-1])
        
    print

File: output/gemma4-e2b/avatar/Python/Java/codeforces_596_B.java
import java.util.Scanner;
import java.lang.Math;

public class codeforces_596_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read n
        if (!scanner.hasNextInt()) {
            return;
        }
        int n = scanner.nextInt();

        // Read the sequence r
        int[] r = new int[n];
        for (int i = 0; i < n; i++) {
            if (!scanner.hasNextInt()) {
                // Handle case where input ends prematurely
                return;
            }
            r[i] = scanner.nextInt();
        }

        if (n == 0) {
            System.out.println(0);
            return;
        }

        // Calculate ans
        // ans = abs(r[0])
        long ans = Math.abs(r[0]);

        // for i in range(1, n): ans += abs(r[i] - r[i - 1])
        for (int i = 1; i < n; i++) {
            ans += Math.abs(r[i] - r[i - 1]);
        }

        System.out.println(ans);
    }
}